loc_books/books.py

```diff
@@ -0,0 +1,213 @@
+        text_file_path = os.path.join(book_downloads_path, text_file_name)
+        if not os.path.exists(text_file_path):
+            try:
+                for attempt in Retrying(
+                    stop=stop_after_attempt(3),
+                    wait=wait_random_exponential(multiplier=1, max=30),
+                ):
+                    with attempt:
+                        try:
+                            response = self.session.get(text_file_url)
+                            response.raise_for_status()
+                            text = response.text
+                            with open(text_file_path, "w") as f:
+                                f.write(text)
+                            return {
+                                "success": True,
+                                "url": text_file_url,
+                                "filename": text_file_name,
+                                "text": text,
+                            }
+                        except requests.exceptions.RequestException as e:
+                            logger.error(
+                                f"Failed to download {text_file_url}: {str(e)}"
+                            )
+                            raise e
+            except RetryError:
+                logger.error(f"Failed to download {text_file_url} after 3 attempts")
+                return {
+                    "success": False,
+                    "url": text_file_url,
+                    "filename": text_file_name,
+                    "text": None,
+                }
+
+        return None
+
+
+class LocBooksExporter:
+    def __init__(self, snapshot):
+        self.snapshot = snapshot
+
+        metadata_path = metadata_exports_path / f"{snapshot}.csv"
+        metadata_file = metadata_path.absolute()
+
+        self.metadata = pd.read_csv(metadata_file)
+        self.metadata["text_file_name"] = self.metadata["text_file_url"].apply(
+            lambda x: Path(str(furl(x).path)).name
+        )
+
+    def export(self, shard_size, filename):
+        text_files = list(book_downloads_path.glob("*.txt"))
+        logger.info(f"Found {len(text_files)} text files to export")
+        results = map(functools.partial(self.format_dolma), text_files)
+
+        export_folder = book_exports_path / self.snapshot / "documents"
+        export_folder.mkdir(parents=True, exist_ok=True)
+
+        to_dolma(results, export_folder, filename, shard_size)
+        logger.info(
+            f"Exported {len(text_files)} text files in dolma format to {export_folder}"
+        )
+
+    def format_dolma(self, filepath):
+        filename = filepath.name
+        metadata = self.metadata.loc[self.metadata["text_file_name"] == filename]
+        if not metadata.empty:
+            with open(filepath) as f:
+                text = f.read()
+
+            dolma_data = {
+                "id": metadata["lccn"].values[0],
+                "text": text,
+                "source": "loc_books",
```

Comment:
I think in other places we use a more verbose source name, like `Library of Congress Books`